### PR TITLE
Feature/206

### LINKED ISSUE #195: Move components to Geppetto-UI
Components signalled as green should be moved into Geppetto-UI folder.
[Spreedsheat](https://docs.google.com/spreadsheets/d/1qQgio7uKc41-h-VoebWW1_qj7Sai07VpiTjh5zu8o2Q/edit#gid=1895320817)

This issue is a subpart of #174 
Branching from [splitRefactoring](github.com/openworm/geppetto-client/tree/splitRefactoring) branch.

### LINKED ISSUE #206: Refactor BigImageViewer Showcase
Create showcase for BigImageViewer and add it to the showcase project.
Additionally adapt the component following the given guidelines:

- move to es6

- replace jquery.extend with new util; 

- remove widget code; 

- set data to props

This issue is a subpart of #174

### PR DESCRIPTION
**Pull request by afonsobspinto**
Closes #195 #197 #202 #206 
Contains:
- The geppetto structural split (client, ui, showcase)
- Showcase for BigImageViewer and Connectivity components, following the enhanced showcase template. See [lunpark-dev#Geppetto-Showcase](https://github.com/MetaCell/lunapark-dev/wiki/Geppetto-Showcase) for additional information.
- Basic showcase for Plot, FlexLayout, MoviePlayer, GraphVisualizer, ListViewer and Menu (all should be 'easily' updatable to the new structure).

Some notes already discussed but will not be on this PR (unless said otherwise):
- Change plot component to use only instances instead of full model
- Python console, show an image (for now)
- Improve showcase UI/UX ( add material paper, add button to copy source code, others...)
- Add toolbar common component
- Do not set the height -> use flex instead

_Originally posted by @filippomc in https://github.com/openworm/geppetto-client/pull/198_:
Those `geppetto-client/geppetto-client` scare me. If someone looks at it without knowing the history easily thinks thinks it's a bad copy/paste to fix. Also we should remove direct references to files being inside node_modules. If we don't do that, we need to include the node_modules folder in every deployment. @tarelli I think that getting rid of those references may be one of the next refactorings.

### CODE REVIEW

**Comment by afonsobspinto** [comment]
Those `geppetto-client/geppetto-client` scare me. If someone looks at it without knowing the history easily thinks thinks it's a bad copy/paste to fix. Also we should remove direct references to files being inside node_modules. If we don't do that, we need to include the node_modules folder in every deployment. @tarelli I think that getting rid of those references may be one of the next refactorings.

_Originally posted by @filippomc in https://github.com/openworm/geppetto-client/pull/198_

File: geppetto-showcase/components/Code.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import IconButtonWithTooltip from "./IconButtonWithTooltip";
+import {faCode} from "@fortawesome/free-solid-svg-icons";
+import Toolbar from "@material-ui/core/Toolbar";
+import SyntaxHighlighter from 'react-syntax-highlighter';
+import { darcula } from 'react-syntax-highlighter/dist/esm/styles/hljs';
+
+const styles = (theme) => ({
+    toolbar: {
+        padding: "0",
+        marginLeft: "5px"
```
Comment (filippomc) [inline geppetto-showcase/components/Code.js:12]: @afonso this is the things we shoulf use spacing for. `theme.spacing(1)`

File: geppetto-showcase/components/Code.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    button: {
+        padding: "8px",
+        top: "0",
+        color: "#fc6320"
```
Comment (filippomc) [inline geppetto-showcase/components/Code.js:17]: color should come from the theme (either primary or secondary)

File: geppetto-showcase/components/Code.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    },
+});
+
+
+class Code extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            source: false,
+            sourceTooltip: "Show the full source code",
```
Comment (filippomc) [inline geppetto-showcase/components/Code.js:27]: Better use constants than string literals

File: geppetto-ui/src/components/big-image-viewer/BigImageViewer.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    bigImageViewer: {
+        height: '800px',
+    },
+    toolbar: {
+        padding: "0",
+        marginLeft: "5px"
+    },
+    toolbarBox: {backgroundColor: "rgb(0,0,0,0.5);",},
+    button: {
+        padding: "8px",
```
Comment (filippomc) [inline geppetto-ui/src/components/big-image-viewer/BigImageViewer.js:21]: theme.spacing

File: geppetto-ui/src/components/big-image-viewer/BigImageViewer.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        top: "0",
+        color: "#fc6320"
```
Comment (filippomc) [inline geppetto-ui/src/components/big-image-viewer/BigImageViewer.js:23]: theme color

File: geppetto-ui/src/components/big-image-viewer/BigImageViewer.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                                disabled={false}
+                                onClick={() => customButton.action()}
+                                className={classes.button}
+                                tooltip={customButton.title}
+                                icon={customButton.icon}
+                            />
+                        )}
+                    </div>
+                </Toolbar>
+                <div id={this.props.id + "_component"} className={classes.bigImageViewer}/>
```
Comment (filippomc) [inline geppetto-ui/src/components/big-image-viewer/BigImageViewer.js:138]: Better avoid ids, too easy to duplicate

File: geppetto-ui/src/components/big-image-viewer/showcase/BigImageViewerConfig.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import BigImageViewerExample
+    from "./examples/BigImageViewerExample";
+
+
+export const BigImageViewerConfig = {
+    'name': "BigImageViewer Main Title",
+    'description': "BigImageViewer Main Description",
```
Comment (filippomc) [inline geppetto-ui/src/components/big-image-viewer/showcase/BigImageViewerConfig.js:7]: Idea for the future: get the description from the readme file inside the component folder

File: geppetto-ui/src/components/flex-layout/style/light.css [hunk 8]
```diff
@@ -139,7 +139,7 @@
 
 .flexlayout__tab_button:hover .flexlayout__tab_button_trailing,
 .flexlayout__tab_button--selected .flexlayout__tab_button_trailing{
-    background: transparent url("../images/close.png") no-repeat center;
+    background: transparent url("geppetto-ui/src/components/flex-layout/images/close.png") no-repeat center;
```
Comment (filippomc) [inline geppetto-ui/src/components/flex-layout/style/light.css:142]: shouldn't refer to source files. Webpack loaders ensure us proper reference in different contexts.

File: geppetto-ui/src/components/movie-player/MoviePlayer.js [hunk 9]
```diff
@@ -2,7 +2,8 @@ define(function (require) {
 
   var React = require('react');
   var ReactPlayer = require('react-player');
-  var AbstractComponent = require('../../AComponent');
+  var AbstractComponent = require('@geppettoengine/geppetto-client/js/components/AComponent');
```
Comment (filippomc) [inline geppetto-ui/src/components/movie-player/MoviePlayer.js:5]: Remove?

File: geppetto-ui/src/components/plot/PlotComponent.js [hunk 10]
```diff
@@ -3,19 +3,15 @@ import JSZip from "jszip";
...
 Plotly.register([require('plotly.js/lib/scatter')]);
 const ScatterPlot = createPlotlyComponent(Plotly);
-
 import { unit } from 'mathjs';
 import PlotHeader from './PlotHeader';
-import AbstractComponent from '../../AComponent';
+import AbstractComponent from '@geppettoengine/geppetto-client/js/components/AComponent';
 import { defaultLayout, defaultTrace, defaultLine, defaultConfig, defaultAxisLayout } from './configuration/plotConfiguration';
-import ExternalInstance from '../../../geppettoModel/model/ExternalInstance';
+import ExternalInstance from '@geppettoengine/geppetto-client/js/geppettoModel/model/ExternalInstance';
```
Comment (filippomc) [inline geppetto-ui/src/components/plot/PlotComponent.js:12]: remove?

File: geppetto-ui/src/components/plot/PlotComponent.js [hunk 11]
```diff
@@ -3,19 +3,15 @@ import JSZip from "jszip";
...
 import Plotly from 'plotly.js/lib/core';
 import createPlotlyComponent from 'react-plotly.js/factory';
-
 Plotly.register([require('plotly.js/lib/scatter')]);
 const ScatterPlot = createPlotlyComponent(Plotly);
-
 import { unit } from 'mathjs';
 import PlotHeader from './PlotHeader';
-import AbstractComponent from '../../AComponent';
+import AbstractComponent from '@geppettoengine/geppetto-client/js/components/AComponent';
```
Comment (filippomc) [inline geppetto-ui/src/components/plot/PlotComponent.js:10]: all references to AbstractComponent should be removed from inside geppetto-ui

File: geppetto-ui/src/components/plot/showcase/PlotShowcase.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    getLegendName (projectId, experimentId, instance, sameProject) {
+        const instancePath = instance.getInstancePath()
+            .split('.')
+            .filter((word, index, arr) => index != 0 && index != arr.length - 1)
+            .join('.');
+
+        if (sameProject) {
+            window.Project.getExperiments().forEach(experiment => {
```
Comment (filippomc) [inline geppetto-ui/src/components/plot/showcase/PlotShowcase.js:23]: Why using experiments to get legend name?

Comment (filippomc) [reply geppetto-ui/src/components/plot/showcase/PlotShowcase.js:23]: We may just use the instance name, or set up a lookup table if we want to be sophisticated

File: geppetto-ui/src/components/big-image-viewer/BigImageViewer.js (hunk 6 above)
Comment (afonsobspinto) [reply geppetto-ui/src/components/big-image-viewer/BigImageViewer.js:113]: I fear it is necessary due to the use of [OpenSeaDragon](https://github.com/openseadragon/openseadragon) (which requires the id of the element where the zoomable image will be added) by the BigImageViewer

Comment (afonsobspinto) [reply geppetto-ui/src/components/big-image-viewer/BigImageViewer.js:113]: Another case where I'm using ids is to add a key to each element of react arrays:
image

**Comment by afonsobspinto** [comment]
Hello @filippomc, I think I addressed all of the minor comments you have added. Please let me know if you want anything else to be changed.

**Pull request merged**